src/components/TradeWidget/TokenRow.tsx

```diff
@@ -146,7 +146,8 @@ const TokenRow: React.FC<Props> = ({
   )
 
   function useMax(): void {
-    setValue(inputId, formatAmountFull(balance.totalExchangeBalance, balance.decimals, false))
+    // boolean true here forces form validation
+    setValue(inputId, formatAmountFull(balance.totalExchangeBalance, balance.decimals, false), true)
```

Comment:
Is that it? add a couple more lines :smile:

Reply:
haha yeah that was it. spent an ungodly amount of time trying other stuff until i utilised the power of types and read the last optional param haha